src/lib/OpenEXRCore/decoding.c

```diff
@@ -351,38 +354,44 @@ unpack_sample_table (
     {
         for (int32_t y = 0; y < h; ++y)
         {
+            int32_t *cursampline = samptable + y * w;
             int32_t prevsamp = 0;
             for (int32_t x = 0; x < w; ++x)
             {
                 int32_t nsamps =
-                    (int32_t) one_to_native32 ((uint32_t) samptable[y * w + x]);
-                if (nsamps < 0) return EXR_ERR_INVALID_SAMPLE_DATA;
-                samptable[y * w + x] = nsamps - prevsamp;
-                prevsamp             = nsamps;
+                    (int32_t) one_to_native32 ((uint32_t) cursampline[x]);
+                if (nsamps < prevsamp) return EXR_ERR_INVALID_SAMPLE_DATA;
+
+                cursampline[x] = nsamps - prevsamp;
+                prevsamp       = nsamps;
             }
-            totsamp += prevsamp;
+            totsamp += (uint64_t)prevsamp;
```

Comment:
is there a reason for prevsamp and nsamps to be signed at all? On line 362 it's getting casted to signed from unsigned, and I can't spot why

Reply:
the specified file type is int32, but the value can't be negative... and was running into wrap-around with malicious files, but could do int64 I suppose.

Reply:
If we assume there are no non-twos-complement computers anymore, can change the internal type to be unsigned, but the exposed count in other places (i.e. C++) is as an int...

Reply:
Well, I buy the malicious wrap-around argument.